packages/@lwc/engine/src/framework/wc.ts

```diff
@@ -25,17 +27,21 @@ import { HTMLElementConstructor } from './base-bridge-element';
  * const elm = document.createElement('x-foo');
  * ```
  */
-export function buildCustomElementConstructor(
-    Ctor: ComponentConstructor,
-    options?: {
-        mode?: 'open' | 'closed';
-    }
+export function deprecatedBuildCustomElementConstructor(
+    Ctor: ComponentConstructor
 ): HTMLElementConstructor {
+    /* eslint-disable-next-line no-console */
+    console.error(
+        'Deprecated function called: "buildCustomElementConstructor" function is deprecated and it will be removed in v1.4.0.' +
+            'Please use "CustomElementConstructor" static property of the component constructor to access the corresponding custom element constructor instead.'
+    );
+
+    return Ctor.CustomElementConstructor;
+}
+
+export function buildCustomElementConstructor(Ctor: ComponentConstructor): HTMLElementConstructor {
     const { props, bridge: BaseElement } = getComponentDef(Ctor);
-    const mode =
-        isUndefined(options) || isUndefined(options.mode) || options.mode !== 'closed'
-            ? 'open'
-            : 'closed';
+    const mode = 'open';
```

Comment:
Inline the `open` in the object literal since it is not used anywhere else in the closure.